src/main/java/org/elasticsearch/common/joda/Joda.java

```diff
@@ -133,6 +134,10 @@ public static FormatDateTimeFormatter forPattern(String input, Locale locale) {
             formatter = ISODateTimeFormat.yearMonth();
         } else if ("yearMonthDay".equals(input) || "year_month_day".equals(input)) {
             formatter = ISODateTimeFormat.yearMonthDay();
+        } else if ("epoch_second".equals(input)) {
```

Comment:
But I think this is confusing. If I have a format specified as `yearMonthDay` that works then I would expect to be able to change it to `epochSecond` and it would work. Supporting some values in camelCase for date formats and not other values is very confusing to a user. I'm all for removing camelCase but we should be consistent with it, especially when its different values for the same setting (in the case different values of `format` for date fields.